### PR TITLE
Add option to format the date output

### LINKED ISSUE #1: Feature:  Flag to format the output
I have a use case for formatted output. And didn’t find a matching CLI for that.
It would be nice this one could handle that.

## Proposal

```
$ date-now --format "DD.MM.YYYY"
25.04.2018

$ date-now --format "YYY:MM:DD"
2018:04:25
```

**Valid patterns**

+ `YYYY`: Full year
+ `MM`: Month (example: 04)
+ `DD`: Day (example: 01)
+ `HH`: Hours (example: 14)
+ `mm`: Minutes (example: 59)
+ `ss`: Seconds (example: 09)
+ `SSS`: Milliseconds (example: 532)

Not sure about the patterns though. It’s a subset auf the format options of:

- [date-fns](https://date-fns.org/v1.29.0/docs/format)
- [Moment.js](https://momentjs.com/docs/#/displaying/format/)

### PR DESCRIPTION
**Pull request by sunilhari**
Hi @sindresorhus 

I was not able to fix the formatting of the code.If you could guide me i can do that.Please review the fix that i have added and kindly guide me for the changes i have to make.

I tried writing a small test case to validate the same.But while running from the testscript the flag option doesnt seem to work.
Guide me if you have some time.
Thanking you in advance.

Fixes #1

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -2,6 +2,13 @@ import test from 'ava';
 import execa from 'execa';
 
 test('main', async t => {
-	const {stdout} = await execa('./cli.js');
+	const { stdout } = await execa('./cli.js');
 	t.regex(stdout, /\d+/);
 });
+/*
+test('Get Formatted Text', async t => {
+	const { stdout } = await execa('./cli.js --format "YYYY"');
```
Comment (sindresorhus) [inline test.js]: You're using the argument format of `execa.shell`. You want to do this instead:

```js
const {stdout} = await execa('./cli.js', '--format', 'YYYY');
```

File: test.js [hunk 2]
```diff
@@ -2,6 +2,13 @@ import test from 'ava';
 import execa from 'execa';
 
 test('main', async t => {
-	const {stdout} = await execa('./cli.js');
+	const { stdout } = await execa('./cli.js');
```
Comment (sindresorhus) [inline test.js]: Don't do unrelated changes.

File: package.json [hunk 3]
```diff
@@ -33,5 +33,8 @@
 		"ava": "*",
 		"execa": "^0.10.0",
 		"xo": "*"
+	},
+	"dependencies": {
```
Comment (sindresorhus) [inline package.json]: Move this before `devDependencies`

File: cli.js [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+
+const DEFAULT_FORMAT = 'x';
+const cli = meow(`
+      Usage
+        $ date-now
+      Options
+        --format <format>, -f Get Formatted Output
+      Examples
+        $ date-now
+        $ date-now --format  "DD.MM.YYYY"
```
Comment (sindresorhus) [inline cli.js]: I think the examples should also show the output

File: cli.js [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+  {
+    flags: {
+      format: {
+        type: 'string',
+        alias: 'f'
+      }
+    }
+  }
+);
+const format = cli.flags.format || DEFAULT_FORMAT;
```
Comment (sindresorhus) [inline cli.js]: When no `format` is not specified, there's no point in running it through `formatter`, so just use `Date.now()` then.

**Comment by sindresorhus** [comment]
Awesome. Thanks for working on this :)

**Comment by sindresorhus** [comment]
Make sure you run `npm test` and that it passes.

File: cli.js [hunk 6]
```diff
@@ -1,3 +1,27 @@
...
+const DEFAULT_FORMAT = 'x';
+const cli = meow(`
+      Usage
+        $ date-now
+      Options
+        --format <format>, -f Get Formatted Output
+      Examples
+        $ date-now
+        $ date-now --format  "DD.MM.YYYY"
+    `,
```
Comment (sindresorhus) [inline cli.js]: We should include a link here at the bottom of the CLI help output to the date-fns format docs.

You could use https://github.com/sindresorhus/terminal-link to make it a real link for supporting terminals.

**Comment by sindresorhus** [comment]
Can you give the PR a more descriptive title?

File: test.js (hunk 1 above)
Comment (sunilhari) [reply test.js]: Hi @sindresorhus 

I tried like this and  did not work.
However this worked
`const {stdout} = await execa('./cli.js', ['--format', 'YYYY']);`

**Comment by sunilhari** [comment]
Hi @sindresorhus 

Kindly review the changes.Thanks for the help and reviews

File: test.js (hunk 1 above)
Comment (sindresorhus) [reply test.js]: Yeah, I forgot the brackets.

**Comment by sindresorhus** [comment]
Thank you, @sunilhari 🙌

**Comment by sunilhari** [comment]
Thank 'You' @sindresorhus
